File: Chain of Responsibility/PHP/Authorization.php
<?php
require_once('Request.php');
require_once('Handler.php');

class Authorization extends Handler {
    function execute(Request $request) {
        if ($request->is_authorized()) {
            echo "Authorized\n";

            return $this->next_handler->execute($request);
        } else {
            echo "Unauthorized\n";

            return new Response(
                403,
                "Unauthorized",
                null
            );
        }
    }
}

File: Chain of Responsibility/Dart/Authorization.dart
import 'Handler.dart';
import 'Response.dart';
import 'Request.dart';

class Authorization extends Handler {
  @override
  Response? execute(Request? request) {
    if (request?.is_authorized == true) {
      print("Authorized");

      return next_handler?.execute(request);
    } else {
      print("Unauthorized");

      return Response(
        403, 
        'Unauthorized', 
        null
      );
    }
  }
}

